test/e2e/specs/site-editor/navigation.spec.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	editorNavigationUtils: async ( { page, pageUtils }, use ) => {
+		await use( new EditorNavigationUtils( { page, pageUtils } ) );
+	},
+} );
+
+test.describe( 'Site editor navigation', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'Can use keyboard to navigate the site editor', async ( {
+		admin,
+		editorNavigationUtils,
+		page,
+		pageUtils,
+	} ) => {
+		await admin.visitSiteEditor();
+
+		// Test: Can navigate to a sidebar item and into its subnavigation frame without losing focus
+		// Go to the Pages button
+
+		await editorNavigationUtils.tabToLabel( 'Pages', { times: 10 } );
+
+		await expect(
+			page.getByRole( 'button', { name: 'Pages' } )
+		).toBeFocused();
+		await pageUtils.pressKeys( 'Enter' );
+		// We should be in the Pages sidebar
+		await expect(
+			page.getByRole( 'button', { name: 'Back', exact: true } )
+		).toBeFocused();
+		await pageUtils.pressKeys( 'Enter' );
+		// Go back to the main navigation
+		await expect(
+			page.getByRole( 'button', { name: 'Pages' } )
+		).toBeFocused();
+
+		// Test: Can navigate into the iframe using the keyboard
+		await editorNavigationUtils.tabToNode( 'IFRAME', { times: 10 } );
+		// Getting the actual iframe as a cleaner locator was suprisingly tricky,
+		// so we're using a css selector with .is-focused which should be present when the iframe has focus.
+		await expect(
+			page.locator( 'iframe[name="editor-canvas"].is-focused' )
+		).toBeFocused();
```

Comment:
```suggestion
		await expect(
			page.getByRole( 'button', { name: 'Editor Canvas' } )
		).toBeFocused();
```

The role of the editor canvas iframe in view mode should be a button, so how about changing it like this?

Reply:
Great idea. That's much better. Thanks!